scalyr_agent/log_processing.py

```diff
@@ -2286,19 +2286,27 @@ def __init__(self, overall_config, log_entry_config, file_system=None):
         # The LogFileProcessor objects for all log files that have matched the log_path.  This will only have
         # one element if it is not a glob.
         self.__processors = []
-        # The lock that protects the __processor, __is_finished and __last_check vars.
+        # The lock that protects the __processor, __is_finishing and __last_check vars.
         self.__lock = threading.Lock()
 
         self.update_log_entry_config( log_entry_config )
 
-        self.__is_finished = False
+        self.__is_finishing = False
         # Determine if the log path to match on is a glob or not by looking for normal wildcard characters.
         # This probably leads to false positives, but that's ok.
         self.__is_glob = '*' in self.log_path or '?' in self.log_path or '[' in self.log_path
         # The time in seconds past epoch when we last checked for new files that match the glob.
         self.__last_check = None
 
     def update_log_entry_config( self, log_entry_config ):
+        """
+        Updates the log config for the log matcher and any of its processors.
+        In order to update the processors, the processors are simply closed and
+        removed from the list of processors.  These processors will be recreated
+        the next time the file matcher is matched.
+
```

Comment:
I would just add some documentation that the caller must retain the checkpoints and be sure they are used when running the matcher the next time.